File: app/src/main/java/com/example/jacob/myapplication/db/DataReaderHelper.java
package com.example.jacob.myapplication.db;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

/**
 * Created by jacob on 22/11/2015.
 */
public class DataReaderHelper extends SQLiteOpenHelper {

    public DataReaderHelper(Context context) {
        super(context, DataReaderContract.DATABASE_NAME, null, DataReaderContract.DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(DataReaderContract.SQL_CREATE_ENTRIES);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL(DataReaderContract.SQL_DELETE_ENTRIES);
        onCreate(db);
    }

    @Override
    public void onDowngrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        onUpgrade(db, oldVersion, newVersion);
    }
}

File: app/src/main/java/me/jacobrr/whatsappanalyzer/db/DataReaderHelper.kt
package me.jacobrr.whatsappanalyzer.db

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

/**
 * Created by jacob on 22/11/2015.
 */
class DataReaderHelper(context: Context) :
    SQLiteOpenHelper(context, DataReaderContract.DATABASE_NAME, null, DataReaderContract.DATABASE_VERSION) {

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(DataReaderContract.SQL_CREATE_ENTRIES)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL(DataReaderContract.SQL_DELETE_ENTRIES)
        onCreate(db)
    }

    override fun onDowngrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        onUpgrade(db, oldVersion, newVersion)
    }
}
